src/GADM.jl

```diff
@@ -52,86 +53,65 @@ function dataread(path)
     !isnothing(data) ? data : throw(ArgumentError("failed to read data from disk"))
 end
 
+"""
+    getdataset(country)
+
+Downloads and extracts dataset of the given country code
+"""
+function getdataset(country)
+    isvalidcode(country) || throw(ArgumentError("please provide standard ISO 3 country codes"))
+    data = country |> download |> dataread
+end
+
 """
     getlayer(data, level) 
 
 Get layer of the desired `level` from the `data`.
 """
 function getlayer(data, level)
     nlayers = ArchGDAL.nlayer(data)
-    for l=0:nlayers-1
+    for l = 0:nlayers - 1
         layer = ArchGDAL.getlayer(data, l)
         lname = ArchGDAL.getname(layer)
         llevel = last(split(lname, "_"))
         string(level) == llevel && return layer
     end
-    throw(ArgumentError("valid levels are 0...$(nlayers-1)"))
+    throw(ArgumentError("asked for level $(level), valid levels are 0-$(nlayers - 1)"))
 end
 
-"""
-    get(country, subregions...)
-
-Returns the MULTIPOLYGON data for the requested region.
-
-Input: ISO 3 country code, and further subdivisions.
-
-## Examples  
-  
-```julia
-get("IND")
-get("IND", "Uttar Pradesh")
-get("IND", "Uttar Pradesh", "Lucknow")
-```
-"""
```

Comment:
So where is this docstring going?

Reply:
I'll add a new docstring in later commits to this PR.